### PR TITLE
MNT: handle warning in scipy cdist for unused parameters

### PR DESCRIPTION
**Pull request by hmaarrfk**
This will be an error in a future version apparently as explained by a warning.

## For reviewers
- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Needs a test to trigger that code path.

File: skimage/feature/match.py [hunk 1]
```diff
@@ -59,7 +59,11 @@ def match_descriptors(descriptors1, descriptors2, metric=None, p=2,
         else:
             metric = 'euclidean'
 
-    distances = cdist(descriptors1, descriptors2, metric=metric, p=p)
+    kwargs = {}
```
Comment (sciunto) [inline skimage/feature/match.py]: @hmaarrfk would you mind fixing the docstring as well? Several missing `, optional` over here...

File: skimage/feature/match.py [hunk 2]
```diff
@@ -59,7 +59,11 @@ def match_descriptors(descriptors1, descriptors2, metric=None, p=2,
...
+    # Scipy raises an error if you pass it an extra argument.
```
Comment (sciunto) [inline skimage/feature/match.py]: I would rephrase as: "Scipy raises an error if p is directly passed as a keyword argument."

File: skimage/feature/tests/test_match.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import numpy as np
-from skimage._shared.testing import assert_equal
+from skimage._shared.testing import assert_equal, parametrize
```
Comment (sciunto) [inline skimage/feature/tests/test_match.py]: I'm missing why this edit is necessary... :/

Comment (hmaarrfk) [reply skimage/feature/tests/test_match.py]: sorry. that was left over from when I was thinking of using parametrize for the tests.

File: skimage/feature/match.py (hunk 2 above)
Comment (hmaarrfk) [reply skimage/feature/match.py]: ok.

File: skimage/feature/match.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/feature/match.py]: ok.
